tests/Enum/LoadEnumToEntityIntegrationTest.php

```diff
@@ -75,23 +75,35 @@ public function testLoadEnumMissingEnumClass(): void
 		$this->callPostLoadEventOnEntity(new BarEntity());
 	}
 
-	public function testLoadEnumNonExistingEnumClass(): void
+	public function loadEnumToNotEnumEntityDataProvider(): Generator
 	{
-		try {
-			$this->callPostLoadEventOnEntity(new BazEntity());
-			Assert::fail('Exception expected');
-		} catch (\Consistence\Doctrine\Enum\NotEnumException $e) {
-			Assert::assertSame('Consistence\Doctrine\Enum\NonExistingClass', $e->getEnumClass());
-		}
+		yield 'non-existing class' => [
+			'entity' => new BazEntity(),
+			'expectedClass' => 'Consistence\Doctrine\Enum\NonExistingClass',
+		];
+
+		yield 'not enum class' => [
+			'entity' => new BaxEntity(),
+			'expectedClass' => FooEntity::class,
+		];
 	}
 
-	public function testLoadEnumNotEnumClass(): void
+	/**
+	 * @dataProvider loadEnumToNotEnumEntityDataProvider
+	 *
+	 * @param \Consistence\Doctrine\Enum\FooParentEntity $notEnumClassEntity
+	 * @param string $expectedClass
+	 */
+	public function testLoadEnumToNotEnumEntity(
```

Comment:
I think the naming here is pretty misleading -`testLoadEnumToNotEnumEntity` - this lookes like it describes loading enum into not an enum - so all cases, because nothing would be an enum and an entity at the same time. Including `FooEntity` for which it would not fail.

So I would either keep the original `testLoadEnumNotEnumClass()`, because it matches the others like `testLoadEnumMissingEnumClass()` etc.?